### PR TITLE
VDB-1368: Add Dockerfile for extract diffs cmd

### PR DESCRIPTION
**Pull request by elizabethengelman**
This PR needs to work with [this VDB PR](https://github.com/makerdao/vulcanizedb/pull/119). Before the VDB PR is merged in, pass `VDB_VERSION=VDB-1368-add-new-geth-patch-as-diff-source` as a build arg while building this container, eg: 

```
docker build -f dockerfiles/extract_diffs/Dockerfile . -t extract --build-arg VDB_VERSION=VDB-1368-add-new-geth-patch-as-diff-source
```

### CODE REVIEW

File: dockerfiles/extract_diffs/startup_script.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+./goose -dir db/migrations postgres "$VDB_PG_CONNECT" up
+
+if [ $? -ne 0 ]; then
+  echo "Could not run migrations. Are the database details correct?"
+  exit 1
+fi
+
+# Run extractDiffs
+echo "Running extractDiffs..."
+./vulcanizedb extractDiffs --watchedAddresses=0x78f2c2af65126834c51822f56be0d7469d7a523e \
```
Comment (elizabethengelman) [inline dockerfiles/extract_diffs/startup_script.sh]: Unfortunately, I don't think that we can add a comment for each address telling which contract it belongs to. Not sure if that's ideal, so I may play around with just using a .toml config file like we're doing with execute.

**Review by yaoandrew — APPROVED** [review]
LGTM - I was able to build the container locally. Of course I didn't read your note at first so I got an error when I tried to run it without `--build-arg` set🤣

**Comment by elizabethengelman** [comment]
I've added extract diffs to the travis deploy script since the last review and would love another set of eyes!

File: environments/extractDiffs.toml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        deployed = 10322821
+    [contract.OASIS_MATCHING_MARKET_ONE]
```
Comment (rmulhol) [inline environments/extractDiffs.toml]: Might be able to remove the Oasis + OSM contracts?

File: dockerfiles/extract_diffs/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+COPY --from=builder /go/src/github.com/makerdao/vdb-mcd-transformers/dockerfiles/extract_diffs/startup_script.sh .
+COPY --from=builder /go/src/github.com/makerdao/vulcanizedb/db/migrations/* db/migrations/
+COPY --from=builder /go/src/github.com/pressly/goose/cmd/goose/goose goose
+# needed for waiting until postgres is ready before starting from docker-compose
+COPY --from=builder /go/src/github.com/makerdao/vulcanizedb/dockerfiles/wait-for-it.sh .
+
+HEALTHCHECK CMD test -f /tmp/connection
+
+# need to execute with a shell to access env variables
+CMD ["./startup_script.sh"]
```
Comment (rmulhol) [inline dockerfiles/extract_diffs/Dockerfile:56]: One thing that bit me recently was forgetting to update permissions on the `startup_script.sh` file - if you haven't run this locally, maybe worth verifying that won't be an issue here

File: .travis/deploy.sh [hunk 4]
```diff
@@ -28,6 +28,9 @@ fi
 message BUILDING EXECUTE DOCKER IMAGE
 docker build -f dockerfiles/execute/Dockerfile . -t makerdao/vdb-execute:$TAG
 
+message BUILDING EXTRACT-DIFFS DOCKER IMAGE
+docker build -f dockerfiles/extract_diffs/Dockerfile . -t makerdao/vdb-extract-diffs:$TAG
```
Comment (rmulhol) [inline .travis/deploy.sh:32]: Since this is moving over here, do we want a corresponding PR to remove it from vdb?

**Review by rmulhol — APPROVED** [review]
A couple questions, but LGTM 👍

File: .travis/deploy.sh (hunk 4 above)
Comment (elizabethengelman) [reply .travis/deploy.sh:32]: 👍 I think that https://github.com/makerdao/vulcanizedb/pull/119 should take care of this.

File: environments/extractDiffs.toml (hunk 2 above)
Comment (elizabethengelman) [reply environments/extractDiffs.toml]: Updated in c0947f8

File: dockerfiles/extract_diffs/Dockerfile (hunk 3 above)
Comment (elizabethengelman) [reply dockerfiles/extract_diffs/Dockerfile:56]: I think this should be okay - I've changed the script's permissions, and can run the docker container on my machine.

**Pull request merged**